Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +15,51 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindDigit(string equation)
+        {
+            int result = -1, defected, defectedIndex;
+            defected = defectedIndex = default(int);
+            string[] components = equation.Split(new[] { '*', '=' }, StringSplitOptions.RemoveEmptyEntries);
+            //storing equation components in separate variables for readibility.
+            string A = components[0];
+            string B = components[1];
+            string C = components[2];
+
+            //Determining the defected component index and missing digit index
+            foreach (var comp in components)
+            {
+                if (comp.IndexOf('?') > -1)
+                {
+                    defectedIndex = comp.IndexOf('?');
+                    defected = Array.IndexOf(components, comp);
+                    break;
+                }
+            }
+
+            //Determining expected value of the defected.
+            int b = (defected != 1) ? int.Parse(B) : int.Parse(A);
+            int c = (defected != 2) ? int.Parse(C) : int.Parse(A);
+            decimal product = (defected == 2) ? b * c : (decimal)c / (decimal)b;
+
+            for (int i = 1; i < 10; i++)

```

Comment:
For is running for 10 times even when the string can be of 4. Iterate it for the char limit of string,

Reply:
New implementation. no loop required.